docs/protocol.md

```diff
@@ -326,142 +326,107 @@ a subsequent visit) some $passphrase$:
-| Return $msg \Vert J \Vert NR$                                                                                 |                         |                                  |
-
-### 11. Source replies to a journalist
-
-See ["Source Submits a Message"](#source-submits-a-message).
+SecureDrop is a first-contact protocol between an unknown party (an anonymous
+source) and well-known parties (journalists).
+
+The preceding setup steps are _role-specific_: sources' and journalists' setup
+steps are different. By contrast, the following protocol steps are
+_role-agnostic_ and _turn-specific_. Except where otherwise noted, sources and
+journalists execute the same fetching step (5), sending step (6), and receiving
+step (7), in any order.
+
+Only a source can initiate a conversation. In other words, a source is always
+the first sender.
+
+### 5. Sender fetches keys and verifies their authenticity <!-- Figure 1 as of 7703a58 -->
+
+**TODO:** Sync with the "Setup" section.
+
+A sender knows their own keys. In addition, in the **reply case,** if the sender
+is a journalist replying to a source, they also already know their recipient's
+keys.
+
+| All senders    | Reply case     |
+| -------------- | -------------- |
+| $pk_S^{APKE}$  | $pk_R^{APKE}$  |
+| $pk_S^{PKE}$   | $pk_R^{PKE}$   |
+| $pk_S^{fetch}$ | $pk_R^{fetch}$ |
+| $sk_S^{APKE}$  |
+| $sk_S^{PKE}$   |
+| $sk_S^{fetch}$ |
+
+For some newsroom $NR$ and all its enrolled journalists $J_i$:
+
+| Sender                                                                                                               |                                 | Server                                                                        |
+| -------------------------------------------------------------------------------------------------------------------- | ------------------------------- | ----------------------------------------------------------------------------- |
+|                                                                                                                      | $\longrightarrow$ `RequestKeys` |                                                                               |
+|                                                                                                                      |                                 | $`pks = \{(pk_{R,i}^{APKE}, pk_{R,i}^{PKE}, pk_{R,i}^{fetch})\}`$ for all $i$ |
+|                                                                                                                      | $`pks \longleftarrow`$          |                                                                               |
+| **TODO:** verification per $NR$ and $J_i$                                                                            |                                 |                                                                               |
```

Comment:
I suggest that we can fill this in once the "Setup" section is updated from #105.